File: src/main/java/net/reflact/client/hud/overlay/TargetHudOverlay.java
package net.reflact.client.hud.overlay;
import net.minecraft.client.MinecraftClient;
import net.minecraft.client.gui.DrawContext;
import net.reflact.client.ReflactClient;
import net.minecraft.entity.LivingEntity;

public class TargetHudOverlay implements ReflactOverlay {
    @Override
    public void render(DrawContext context, float tickDelta) {
        MinecraftClient client = MinecraftClient.getInstance();
        if (client.targetedEntity instanceof LivingEntity livingTarget) {
             int x = getX();
             int y = getY();
             context.drawText(client.textRenderer, livingTarget.getDisplayName(), x, y, 0xFFFFFFFF, true);
             float health = livingTarget.getHealth();
             float maxHealth = livingTarget.getMaxHealth();
             float percent = health / maxHealth;
             int targetBarWidth = 100;
             int targetBarHeight = 8;
             int targetBarY = y + 12;
             context.fill(x, targetBarY, x + targetBarWidth, targetBarY + targetBarHeight, 0xFF000000);
             int fill = (int)(targetBarWidth * percent);
             if (fill > 0) context.fill(x, targetBarY, x + fill, targetBarY + targetBarHeight, 0xFFFF0000);
             String targetHpText = String.format("%.0f/%.0f", health, maxHealth);
             context.drawText(client.textRenderer, targetHpText, x + 2, targetBarY, 0xFFFFFFFF, true);
        }
    }
    @Override public int getX() { return ReflactClient.CONFIG.targetHudX(); }
    @Override public int getY() { return ReflactClient.CONFIG.targetHudY(); }
    @Override public void setX(int x) { ReflactClient.CONFIG.targetHudX(x); }
    @Override public void setY(int y) { ReflactClient.CONFIG.targetHudY(y); }
    @Override public int getWidth() { return 100; }
    @Override public int getHeight() { return 30; }
    @Override public boolean isEnabled() { return ReflactClient.CONFIG.showTargetHud(); }
    @Override public String getName() { return "Target HUD"; }
}

File: src/main/kotlin/net/reflact/client/hud/overlay/TargetHudOverlay.kt
package net.reflact.client.hud.overlay

import net.minecraft.client.MinecraftClient
import net.minecraft.client.gui.DrawContext
import net.minecraft.entity.LivingEntity
import net.reflact.client.ReflactClient

class TargetHudOverlay : ReflactOverlay {
    override fun render(context: DrawContext, tickDelta: Float) {
        val client = MinecraftClient.getInstance()
        val target = client.targetedEntity
        if (target is LivingEntity) {
            val x = x
            val y = y
            context.drawText(client.textRenderer, target.displayName, x, y, 0xFFFFFFFF.toInt(), true)
            val health = target.health
            val maxHealth = target.maxHealth
            val percent = health / maxHealth
            val targetBarWidth = 100
            val targetBarHeight = 8
            val targetBarY = y + 12

            context.fill(x, targetBarY, x + targetBarWidth, targetBarY + targetBarHeight, 0xFF000000.toInt())
            val fill = (targetBarWidth * percent).toInt()
            if (fill > 0) context.fill(x, targetBarY, x + fill, targetBarY + targetBarHeight, 0xFFFF0000.toInt())

            val targetHpText = String.format("%.0f/%.0f", health, maxHealth)
            context.drawText(client.textRenderer, targetHpText, x + 2, targetBarY, 0xFFFFFFFF.toInt(), true)
        }
    }

    override var x: Int
        get() = ReflactClient.CONFIG.targetHudX()
        set(value) { ReflactClient.CONFIG.targetHudX(value) }
    override var y: Int
        get() = ReflactClient.CONFIG.targetHudY()
        set(value) { ReflactClient.CONFIG.targetHudY(value) }
    override var width: Int get() = 100; set(value) {}
    override var height: Int get() = 30; set(value) {}
    override val isEnabled: Boolean get() = ReflactClient.CONFIG.showTargetHud()
    override val name: String get() = "Target HUD"
}
